identity/identity.go

```diff
@@ -103,66 +144,179 @@ func GetIdentityHeader(ctx context.Context) string {
 	return ""
 }
 
-func checkHeader(id *XRHID, w http.ResponseWriter) error {
+// EnforceIdentity extracts, checks and places the X-Rh-Identity header into the
+// request context. If the Identity is invalid, the request will be aborted.
+// No logging is performed, errors are returned with HTTP code 400 and plain
+// text in response body. For more control of the logging or payload, use
+// DecodeAndCheckIdentity and DecodeIdentityCtx exported functions and write
+// your own middleware.
+//
+// Deprecated in v2, use EnforceIdentityWithLogger.
+func EnforceIdentity(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctx, err := DecodeIdentityCtx(r.Context(), r.Header.Get("X-Rh-Identity"))
+		if err != nil {
+			http.Error(w, http.StatusText(400)+": "+err.Error(), 400)
+			return
+		}
+		next.ServeHTTP(w, r.WithContext(ctx))
+	})
+}
+
+// GetIdentity returns the identity struct from the context or empty value when not present.
+func GetIdentity(ctx context.Context) XRHID {
+	value := ctx.Value(parsedKey)
+	if value == nil {
+		return XRHID{}
+	}
+	return value.(XRHID)
```

Comment:
How confident are you that this won't panic the program? Wouldn't this be safer?

```suggestion
	if v, ok := value.(XRHID); ok {
	    return v
	} else {
	    return XRHID{}
	}
```

Reply:
Pretty confident because the context key type named `parsedKey` is now not exported, thus the only code that can insert such value into a context is few lines below: `context.WithValue(ctx, parsedKey, id)` and you cannot pass `nil` into this function because `id` is not a pointer.

Reply:
No, I'm not going to insist. You make a reasonable argument for the decision to explicitly unwrap the value as an `XRHID` type. I agree the most likely scenario encountered here is the absence of the header, which you account for by checking whether the value is nil already.